README.md

```diff
@@ -1,22 +1,37 @@
 # cosmogony
 
-:construction::warning: This is a work in progress, for the moment we have ideas but no code :warning::construction:
+:construction::warning: This is a work in progress. Take a look at the issues if you want to contribute :warning::construction:
 
-##  Goals
 The goal of the project is to have easy to use, easy to update geographic regions.
 
-[OpenStreetMap](https://www.openstreetmap.org) (OSM) seems the best datasource for this, but the OSM administrative regions (admins) have several drawbacks.
+It provides geographical zones with a structured hierarchy to easily know that [Paris](https://www.openstreetmap.org/relation/7444) is `city` in the `state` [Île-de-France](https://www.openstreetmap.org/relation/8649) in the `country` [France](https://www.openstreetmap.org/relation/2202162).
 
- * admin_level : The world is a complicated place, and each country has its own administrative division. OSM uses an `admin_level` tag, with values from 1 to ~10 to allow consistent rendering of the borders among countries. This is fine for making maps, but if you want a world list of cities or regions, you still need local and specific knowledge to find which admin_level to use in each country.
- * no hierarchy (:construction:)
+The general idea of the project is to take OpenStreetMap data and:
+ * use the [well defined libpostal rules](https://github.com/openvenues/libpostal/tree/master/resources/boundaries/osm) to type each zone according to its country
+ * use geographic inclusion to build a hierarchy
 
-We need a structured admin hierarchy to easily know that [paris](https://www.openstreetmap.org/relation/7444) is `city` in the `state` [ile de france](https://www.openstreetmap.org/relation/8649) in the `country` [france](https://www.openstreetmap.org/relation/2202162).
+## Use
 
-OSM administrative regions may not be mapped with the same precision all over the earth but the data is easy to update and the update will benefit the community.
+### Get data
+:construction: We may provide direct data download in the future. For now, you have to extract your geographic regions by yourself :construction:
 
-We do not forbid ourself however to use other data sources (with compliant license), but we don't want `cosmogony` to be too complex and we do not aim to recreate the great [WhosOnFirst](https://www.whosonfirst.org/) ([see below](#See-also))
+#### Build
+You will need
+* rust (`curl https://sh.rustup.rs -sSf | sh`)
+* GEOS (`apt-get install libgeos-dev`)
+
+Then, build cosmogony: `cargo build --release`
+
+You can now grab some OSM pbf and extract your geographic zones:
 
-## Use cases
+`./target/release/cosmogony -i /path/to/your/file.osm.pbf --libpostal /path/to/libpostal/rules/folder/ -o /path/for/output/file`
```

Comment:
it's better to use `cargo run --release`

`cargo run --release -- -i /path/to/your/file.osm.pbf --libpostal /path/to/libpostal/rules/folder/ -o /path/for/output/file`